### PR TITLE
Remove NotImplementedError for parse_dates keyword in read_excel

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Rebase and update of PR https://github.com/pydata/pandas/pull/12051

### CODE REVIEW

File: doc/source/io.rst [hunk 1]
```diff
@@ -2503,6 +2503,21 @@ indices to be parsed.
...
+
+Datetime-like values are automatically converted to the appropriate dtype when
+reading the excel file. When there is a column of strings that have to be parsed
+to a datetime, you can use the `parse_dates` keyword:
+
+.. code-block:: python
+
+   read_excel('path_to_file.xls', 'Sheet1', parse_dates=['date_strings'])
+
+So this should *not* be used when the column in excel has already a datetime-like
```
Comment (jreback) [inline doc/source/io.rst]: small nit. I would the last part to the first paragraph you added. 

somethings like. normally dates are automatically parsed. But if you have a column of 'strings' that *look-like* dates, you can use the parse_dates kw.

**Comment by jreback** [comment]
lgtm.

File: doc/source/io.rst (hunk 1 above)
Comment (jorisvandenbossche) [inline doc/source/io.rst]: yep, adapted a bit

**Comment by jorisvandenbossche** [comment]
Apparently it does not work out of the box anymore by re-enabling it (probably due to changes in `PythonParser` over the last year), so will have to take a deeper look

**Comment by jreback** [comment]
@jorisvandenbossche ?

**Comment by jreback** [comment]
@jorisvandenbossche let's update and get this in.

**Comment by jreback** [comment]
@jorisvandenbossche can you update

**Comment by jreback** [comment]
@jorisvandenbossche can you rebase

**Pull request closed without merging**